pkg/sql/sem/builtins/generator_builtins.go

```diff
@@ -239,23 +317,19 @@ func (s *seriesValueGenerator) Close() {}
 
 // Next implements the tree.ValueGenerator interface.
 func (s *seriesValueGenerator) Next() (bool, error) {
-	if !s.nextOK {
-		return false, nil
-	}
-	if s.step < 0 && (s.start < s.stop) {
-		return false, nil
-	}
-	if s.step > 0 && (s.stop < s.start) {
-		return false, nil
-	}
-	s.value = s.start
-	s.start, s.nextOK = tree.AddWithOverflow(s.start, s.step)
-	return true, nil
+	return s.next(s)
 }
 
 // Values implements the tree.ValueGenerator interface.
 func (s *seriesValueGenerator) Values() tree.Datums {
-	return tree.Datums{tree.NewDInt(tree.DInt(s.value))}
+	switch v := s.value.(type) {
```

Comment:
It would be better to switch on genType than the value. Some values might have the same underlying go type.

Reply:
I change the logic, and make genValue as a member instead of using switch